Explorer/Assets/DCL/Backpack/Gifting/Presenters/Grid/EmoteGridPresenter.cs

```diff
@@ -0,0 +1,116 @@
+using UnityEngine;
+using UnityEngine.Pool;
+using Utility;
+
+namespace DCL.Backpack.Gifting.Presenters.Grid
+{
+    public class EmoteGridPresenter : GiftingGridPresenterBase<EmoteViewModel>
+    {
+        private readonly IEmoteProvider emoteProvider;
+        private readonly IWeb3IdentityCache identityCache;
+        private readonly IWearableStylingCatalog stylingCatalog;
+        private readonly IEmoteProvider.OrderOperation currentOrder = new("date", isAscendent: false);
+
+        public EmoteGridPresenter(
+            GiftingGridView view,
+            SuperScrollGridAdapter<EmoteViewModel> adapter,
+            IEventBus eventBus,
+            LoadGiftableItemThumbnailCommand loadThumbnailCommand,
+            IAvatarEquippedStatusProvider equippedStatusProvider,
+            IPendingTransferService pendingTransferService,
+            IEmoteProvider emoteProvider,
+            IWeb3IdentityCache identityCache,
+            IWearableStylingCatalog stylingCatalog,
+            IWearableStorage wearableStorage,
+            IEmoteStorage  emoteStorage)
+            : base(view,
+                adapter,
+                eventBus,
+                loadThumbnailCommand,
+                equippedStatusProvider,
+                pendingTransferService,
+                wearableStorage,
+                emoteStorage)
+        {
+            this.emoteProvider = emoteProvider;
+            this.identityCache = identityCache;
+            this.stylingCatalog = stylingCatalog;
+
+            adapter.UseWearableStyling(stylingCatalog);
+        }
+
+        protected override async UniTask<(IEnumerable<IGiftable> items, int total)> FetchDataAsync(int pageItemCount, int page, string search, CancellationToken ct)
+        {
+            ReportHub.Log(ReportCategory.GIFTING, $"[Gifting-Emotes] Request Page {page} search='{search}'");
+
+            using var pageEmotesScope = ListPool<IEmote>.Get(out var pageEmotes);
+
+            var requestOptions = new IEmoteProvider.OwnedEmotesRequestOptions(
+                pageNum: page,
+                pageSize: pageItemCount,
+                collectionId: null,
+                orderOperation: currentOrder,
+                name: search,
+                includeAmount: true);
+
+            int total = await emoteProvider.GetOwnedEmotesAsync(
+                identityCache.Identity!.Address,
+                ct,
+                requestOptions,
+                pageEmotes);
+
+            var giftables = new List<IGiftable>(pageEmotes.Count);
+            foreach (var e in pageEmotes)
+                giftables.Add(new EmoteGiftable(e));
+
+            return (giftables, total);
+        }
+
+        protected override EmoteViewModel CreateViewModel(IGiftable item, int amount, bool isEquipped, bool isGiftable)
+        {
+            // Safe cast because FetchDataAsync only produces EmoteGiftable
+            return new EmoteViewModel((EmoteGiftable)item, amount, isEquipped, isGiftable);
```

Comment:
`Amount` is part of `IAvatarAttachment` so there should be no need to cast it to the specific type if you follow my other pieces of advice